### PR TITLE
Tabs: Update subcomponents to accept full HTML element props

### PR DESCRIPTION
**Pull request by chad1008**
Related [comment](https://github.com/WordPress/gutenberg/pull/53960#pullrequestreview-1661817575).

## What?
Updates the subcomponents (TabList, Tab, and TabPanel) to accept all of the HTML element props relevant to their underlying DOM elements.

This PR also removes some artifacts of an old `icon` prop that was removed earlier in the component's development. There were still some references in the types and tests files that needed to be removed.

## Why?
Previously the Tabs subcomponents explicitly accepted props like `className` and `style`, but not the general props for their respective DOM elements. This made it hard to pass basic props, and made it necessary to overuse the `render` prop to gain access to them ([example](https://github.com/WordPress/gutenberg/pull/55360#discussion_r1381348291))

## How?
Each of the subcomponents now leverages `WordPressComponentProps` to include all of the relevant HTML element props.

## Testing Instructions
1. Open the `Tabs` default story in Storybook.
2. Edit the default story by adding some `div` related HTML props to a `TabList` and `TabPanel` subcomponent (`hidden`, `style`, `className` for example, but experiment with any prop that's valid on a `div`.)
3. Confirm that the props you've added are correctly passed to the rendered `div` for the subcomponent you modified.
4. Edit the default story by adding some `button` related props to a `Tab` subcomponent. Specifically, test an `onClick` to confirm that it's applied and functions correctly.
5. Have a snack. You deserve it.

### CODE REVIEW

**Review by brookewp — APPROVED** [review]
This is testing well! ✅ 

Since props are being removed (`className` and `style`), I wonder if it would be beneficial to add any tests? But since this is utilizing `WordPressComponentProps`, it may be unnecessary. I'll leave that up to your discretion! Otherwise, I believe this just needs a CHANGELOG, and then it'll be ready to go.

**Comment by chad1008** [comment]
Thanks @brookewp! I think I'll hold off on tests now... I feel like I'd be writing tests for `WordPressComponentProps` more than for `Tabs` itself - but I'm happy to add them in a followup later if we decide it's warranted!

**Pull request merged**
